akka-docs/src/main/paradox/stream/stream-rate.md

```diff
@@ -203,28 +203,48 @@ Scala
 Java
 :   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #conflate-sample }
 
-### Understanding expand
+### Understanding extrapolate and expand
 
-Expand helps to deal with slow producers which are unable to keep up with the demand coming from consumers.
-Expand allows to extrapolate a value to be sent as an element to a consumer.
+Now we will discuss two stages, `extrapolate` and `expand`, helping to deal with slow producers that are unable to keep up with the demand coming from consumers.
+They allow for additional values to be sent as elements to a consumer.
 
-As a simple use of `expand` here is a flow that sends the same element to consumer when producer does not send
-any new elements.
+As a simple use case of `extrapolate`, here is a flow that repeats the last emitted element to a consumer, whenever the consumer signals demand and the producer cannot supply new elements yet.
 
 Scala
-:   @@snip [RateTransformationDocSpec.scala]($code$/scala/docs/stream/RateTransformationDocSpec.scala) { #expand-last }
+:   @@snip [RateTransformationDocSpec.scala]($code$/scala/docs/stream/RateTransformationDocSpec.scala) { #extrapolate-last }
 
 Java
-:   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #expand-last }
+:   @@snip [RateTransformationDocTest.java]($code$/java/jdocs/stream/RateTransformationDocTest.java) { #extrapolate-last }
 
-Expand also allows to keep some state between demand requests from the downstream. Leveraging this, here is a flow
-that tracks and reports a drift between fast consumer and slow producer.
+For situations where you have several producers, and the entire stream may become blocked by a non-emitting one, you can use the `initial` parameter of `extrapolate` to "seed" the stream.
```

Comment:
Same here as above, "For situations where there may be downstream demand before any element is emitted from upstream", it's not really related to "several producers"

Reply:
You're right, that's only an example scenario, not a description of the general problem case.